libs/tup-components/src/allocations/AllocationTable.test.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { AllocationsTable } from './AllocationTable';
+import { server, testRender } from '@tacc/tup-testing';
+import { screen, waitFor } from '@testing-library/react';
+import { rest } from 'msw';
+
+describe('Projects Allocation Table', () => {
+  it('should display a spinner while loading', async () => {
+    const { getByTestId } = testRender(<AllocationsTable />);
+    expect(getByTestId('loading-spinner')).toBeDefined();
+  });
+  it('should display an error message if an error is returned', async () => {
+    server.use(
+      rest.get('http://localhost:8001/projects', (req, res, ctx) =>
+        res.once(ctx.status(404))
+      )
+    );
+    testRender(<AllocationsTable />);
+    await screen.findAllByText(/Unable to retrieve allocation information./);
+  });
+  it('should display project allocation information', async () => {
+    const { getByText, getByTestId, getAllByRole } = testRender(
+      <AllocationsTable />
+    );
+    await waitFor(() => getAllByRole('columnheader'));
+    const columnHeaders: HTMLElement[] = getAllByRole('columnheader');
```

Comment:
**Reminder to Self:** This will work even without the `role` attribute. ["Default roles are taken into consideration"](https://testing-library.com/docs/queries/byrole/#api).